drivers: sensor: Convert fxos8700 Kconfigs to dts properties

MaureenHelm opened the pull request:
This lays some groundwork for supporting multiple instances of the fxos8700 driver with different properties on each instance. We convert pulse detection, range, and power mode options from Kconfigs to optional device tree properties.

I'm not sure yet how to convert `FXOS8700_DRDY_INT1`, `FXOS8700_PULSE_INT1`, and `FXOS8700_MOTION_INT1`. We probably need to add support for using both INT1 and INT2 simultaneously (i.e., data ready interrupt on INT1 and pulse interrupt on INT2).

I also need to figure out how to handle the different modes (accelerometer-only, magnetometer-only, and hybrid accel+mag).

Would like to get some feedback to see if folks agree with this direction before I continue further.

MaureenHelm committed:
drivers: sensor: Convert fxos8700 pulse Kconfigs to dts properties

Converts fxos8700 pulse detection options from Kconfigs to optional
device tree properties.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
drivers: sensor: Convert fxos8700 range Kconfigs to dts property

Converts fxos8700 range options (2g, 4g, 8g mode) from Kconfigs to an
optional device tree property.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
drivers: sensor: Convert fxos8700 power mode Kconfigs to dts property

Converts fxos8700 power mode options (normal, low noise low power, high
resolution, low power) from Kconfigs to an optional device tree
property.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

gmarull commented on dts/bindings/sensor/nxp,fxos8700.yaml:
```diff
@@ -31,3 +31,89 @@ properties:
+        - 0 # Normal
+        - 1 # Low noise, low power
+        - 2 # High resolution
+        - 3 # Low power
```
It would probably be good to also introduce some device tree header with definitions for that.

gmarull commented on drivers/sensor/fxos8700/fxos8700.c:
```diff
@@ -578,13 +567,13 @@ static const struct fxos8700_config fxos8700_config = {
-	.pulse_tmlt = CONFIG_FXOS8700_PULSE_TMLT,
-	.pulse_ltcy = CONFIG_FXOS8700_PULSE_LTCY,
-	.pulse_wind = CONFIG_FXOS8700_PULSE_WIND,
+	.pulse_cfg = DT_INST_PROP_OR(0, pulse_cfg, 0x3f),
```
Defaults may be placed in binding yaml file

gmarull reviewed (commented):
Driver should also be changed to be multi-instance (remove all `DT_INST(0...` dependencies).

jfischer-phytec-iot commented:
> We convert pulse detection, range, and power mode options from Kconfigs to optional device tree properties.

Looks good, except that someone might want to change it at runtime, but that was not possible even with Kconfig.

> I'm not sure yet how to convert `FXOS8700_DRDY_INT1`, `FXOS8700_PULSE_INT1`, and `FXOS8700_MOTION_INT1`. We probably need to add support for using both INT1 and INT2 simultaneously (i.e., data ready interrupt on INT1 and pulse interrupt on INT2).

I would leave it as it is for now.

MaureenHelm committed:
drivers: sensor: Convert fxos8700 vector mag Kconfigs to dts properties

Converts fxos8700 magnetic vector magnitude options from Kconfigs to
optional device tree properties.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
drivers: sensor: Fix fxos8700 device driver_api/data attributes

Fixes the fxos8700 sensor driver to reflect recent changes in the
generic device driver structure, renaming driver_data and config_info to
data and config respectively. These instances in the fxos8700 sensor
driver were missed because we haven't enabled the magnetic vector
magnitude function in CI.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
samples: sensor: Enable fxos8700 magnetic vector magnitude function

Enables the magnetic vector magnitude function in the fxos8700 sensor
sample application to ensure it builds in CI.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
drivers: sensor: Rework fxos8700 driver to use DT_INST_FOREACH

Reworks the fxos8700 sensor driver to use DT_INST_FOREACH_STATUS_OKAY.
This allows it to support multiple instances, however only a single
instance was tested with the frdm_k64f board.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm commented:
@gmarull @jfischer-phytec-iot please have another look. I rebased to the latest master, moved default values from the driver to the dts bindings, and converted the driver to use `DT_INST_FOREACH_STATUS_OKAY`.

gmarull commented on dts/bindings/sensor/nxp,fxos8700.yaml:
```diff
@@ -126,3 +126,23 @@ properties:
+      required: false
+      default: 0x00
+      description: Magnetic vector-magnitude threshold most significant byte.
+        Resolution is 0.1 Î¼T/LSB.
```
` Î¼T/LSB.` copy/paste artifact?, I suppose it should be `uT/LSB`

MaureenHelm replied:
Good catch, fixed.

gmarull commented on dts/bindings/sensor/nxp,fxos8700.yaml:
```diff
@@ -126,3 +126,23 @@ properties:
+      required: false
+      default: 0x5a
+      description: Magnetic vector-magnitude threshold least significant byte.
+        Resolution is 0.1 Î¼T/LSB.
```
same here

gmarull reviewed (approved):
LGTM, added minor comments.

MaureenHelm commented:
@jfischer-phytec-iot ping
